doc/source/management/data_at_rest_encryption.rst

```diff
@@ -0,0 +1,213 @@
+using `--early-plugin-load variable
+<https://dev.mysql.com/doc/refman/5.7/en/server-options.html#option_mysqld_early-plugin-load>`_
+option:
+
+.. code-block:: bash
+
+  --early-plugin-load="keyring_vault=keyring_vault.so" \
+  --loose-keyring_vault_config="/home/mysql/keyring_vault.conf"
+
+It should be loaded this way to be able to facilitate recovery for encrypted
+tables. However, if you do not care about that you can also use
+``INSTALL PLUGIN`` command:
+
+.. code-block:: mysql
+
+  mysql> INSTALL PLUGIN keyring_vault SONAME 'keyring_vault.so';
+
+Apart from installing plugin you also need to set the
+:variable:`keyring_vault_config` variable. This variable should point to the
+keyring_vault configuration file.
+
+This plugin supports the SQL interface for keyring key management described in
+`General-Purpose Keyring Key-Management Functions
+<https://dev.mysql.com/doc/refman/5.7/en/keyring-udfs-general-purpose.html>`_
+manual.
+
+To enable the functions you'll need to install the ``keyring_udf`` plugin:
+
+.. code-block:: mysql
+
+  mysql> INSTALL PLUGIN keyring_udf SONAME 'keyring_udf.so';
+
+Usage
+-----
+
+On plugin initialization ``keyring_vault`` connects to the Vault server using
+credentials stored in the credentials file. Location of this file is specified
+in by :variable:`keyring_vault_config`. On successful initialization it
+retrieves keys signatures and stores them inside an in-memory hash map.
+
+Configuration file should contain following information:
+
+* ``vault_url`` - the address of the server where Vault is running. It can be a
+  named address – like in example or ip address. The important part is that
+  it should start with ``https://``
+
+* ``secret_mount_point`` - the name of the mount point where ``keyring_vault``
+  will store keys
+
+* ``token`` - the token that ``keyring_vault`` should use when connecting to
+  the Vault. When ``keyring_vault`` is to used only for transparent data
+  encryption and not for ``keyring_udf`` plugin, the token allows for storing
+  new keys in secret mount point. However if ``keyrning_udf`` plugin is used
+  with ``keyring_vault`` plugin this token also allows removing keys from
+  vault. This is due to the fact that ``keyring_udf`` also support
+  ``keyring_key_remove`` operation.
+
+* ``vault_ca [optional]`` - this variable needs to be specified only when
+  Vault's CA certificate is not trusted by the machine that is going to connect
+  to the Vault server. In this case this variable should point CA certificate
+  that was used to signed Vault’s certificates.
+
+Example of the configuration file looks like this: ::
+
+  vault_url = https://vault.public.com:8202
+  secret_mount_point = secret
+  token = 58a20c08-8001-fd5f-5192-7498a48eaf20
+  vault_ca = /data/keyring_vault_confs/vault_ca.crt
+
+When a key is fetched from a ``keyring`` for the first time ``keyring_vault``
+goes to the Vault server and retrieves key's type and data. Next it stores
```

Comment:
"queries the Vault server for the key type and data and caches it locally", replacing the next sentence too